blog/docs/articles/knative-backstage-security.md

```diff
@@ -0,0 +1,60 @@
+# Knative Backstage Security
+
+**Author: Ali Ok, Principal Software Engineer @ Red Hat**
+
+## What's new?
+
+In a previous [blog post](../knative-backstage-plugins/), we talked about how to integrate Knative with Backstage. In this blog post, we will talk about how to secure the communication between the Knative Event Mesh plugin and the backend.
+
+Previously, the backend was running with a service account that had full access to the Kubernetes cluster. This was not secure, as the backend could access any resource in the cluster. Also, the backend didn't have any authentication mechanism, so anyone who could access the backend could access the Kubernetes resources, although they were only read-only.
+
+To solve these issues, we have done 2 things:
+
+1. The backend now uses a service account with limited permissions.
+2. The backend now requires a token to authenticate (passing it along to the API server), for each request coming from the plugin.
+
+## How it works?
+
+![](/blog/articles/images/knative-backstage-security-01.png)
+*Backstage Security*
+[//]: # (https://drive.google.com/file/d/1qMu0yd-zGYcveUM_tLigw1yZ_0jksX9i/view?usp=drive_link)
+
+Similar to other Backstage plugins, we wanted the plugin administrator to configure the plugin by setting up the necessary things like the backend URL and the token. It is a similar approach with the [Backstage Kubernetes plugin](https://backstage.io/docs/features/kubernetes/configuration#configuring-kubernetes-clusters), where the user needs to provide the URL and the token.
+
+The token is stored in Backstage configuration and is passed to the backend with each request. The backend uses this token to authenticate to the Kubernetes API server. The token is a service account token that has the necessary permissions to list the Knative Eventing resources in the cluster.
+
+```yaml
+...
+catalog:
+  providers:
+    knativeEventMesh:
+      dev:
+        token: "eyJhb.........NHA"
```

Comment:
I like `token: '${KNATIVE_EVENT_MESH_TOKEN}' ` better (as in the README).

I'd also add a note here, to not hard-code anything.

Reply:
@matzew new look?